plugins/plugin-defs.nix

```diff
@@ -84,4 +84,42 @@
       sha256 = "sha256-AtkG2XRVZgvJzH2iLr7UT/U1+LXxenvNckdapnJV+8A=";
     };
   };
+
+  magma-nvim = pkgs.vimUtils.buildVimPlugin rec {
+    pname = "magma-nvim";
+    version = "94370733757d550594fe4a1d65643949d7485989";
+
+    src = pkgs.fetchFromGitHub {
+      # using fork of https://github.com/dccsillag/magma-nvim because there are some unmerged
+      # pull requests that I really want
+      owner = "WhiteBlackGoose";
+      repo = "magma-nvim-goose";
+      rev = version;
+      sha256 = "sha256-IaslJK1F2BxTvZzKGH9OKOl2RICi4d4rSgjliAIAqK4=";
+    };
```

Comment:
Not entirely certain about using the fork for the core here, might make more sense to use the default and provide a way to override the package.

Typically, for nixpkgs modules, this is done by adding a "pkg" attribute to the options with a default of the package you want. With that, you can change the package used while still providing a "normal" default for most users.

Reply:
That's a good point, and also having the `package` option is something that's (slowly) being implemented (see #58)